evm_loader/program/src/evm/opcode.rs

```diff
@@ -1304,12 +1304,11 @@ impl<B: Database> Machine<B> {
     #[maybe_async]
     pub async fn opcode_return_impl(
         &mut self,
-        mut return_data: Buffer,
+        return_data: Buffer,
         backend: &mut B,
     ) -> Result<Action> {
         if self.reason == Reason::Create {
-            let code = std::mem::take(&mut return_data);
-            backend.set_code(self.context.contract, code)?;
```

Comment:
Rejected based on this.
`return_data` size can be up to 24kb.
When executing in Solana, we don't have enough memory to copy this.

Reply:
I made `Buffer::clone` cheap for `Inner::Owned` case by using a `Rc<>` to avoid copying the internal buffer when cloning.

Reply:
I extracted this change in #227 for further discussions.